### PR TITLE
LAPP-21 and LAPP-22

### PR DESCRIPTION
**Pull request by cgalan-applaudostudios**
The user-agent-header option should be configurable.

**The backoff policy should be:**  
Initial backoff is 100 milliseconds
Backoff increases by a factor of 2 between intervals 
Backoff introduces some jitter

### CODE REVIEW

File: Analytics/Defaults.cs [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
-using System.Linq;
-using System.Text;
+using Segment.Model;
 
 namespace Segment
 {
     public class Defaults
     {
-        public static string Host = "https://api.segment.io";
+        public static readonly string Host = "https://api.segment.io";
```
Comment (lubird) [inline Analytics/Defaults.cs]: Why did you change these to `readonly`?

File: Analytics/Request/BlockingRequestHandler.cs [hunk 2]
```diff
@@ -90,17 +94,19 @@ internal class BlockingRequestHandler : IRequestHandler
...
 
-        internal BlockingRequestHandler(Client client, TimeSpan timeout, IHttpClient httpClient)
+        internal BlockingRequestHandler(Client client, TimeSpan timeout, IHttpClient httpClient, Backo backo)
 #else
-        internal BlockingRequestHandler(Client client, TimeSpan timeout, HttpClient httpClient)
+        internal BlockingRequestHandler(Client client, TimeSpan timeout, HttpClient httpClient, Backo backo)
 #endif
         {
             this._client = client;
+            _backo = backo;
```
Comment (lubird) [inline Analytics/Request/BlockingRequestHandler.cs]: What happens if `backo` isn't defined? Are we requiring `backo` now in the interface? Or can it be optional still?

File: Analytics/Defaults.cs (hunk 1 above)
Comment (cgalan-applaudostudios) [reply Analytics/Defaults.cs]: Because those are the defaults value and I think that the final user shouldn't be able to change it.

File: Analytics/Request/BlockingRequestHandler.cs (hunk 2 above)
Comment (cgalan-applaudostudios) [reply Analytics/Request/BlockingRequestHandler.cs]: if backo isn't defined an exception is thrown. Backo is required because it implements the retry strategy, but these constructors are internal, this means that the final user doesn't have access to them. The client class uses the constructor with a default instantiation of backo (10s max and 5s of jitter), the other constructor was added for testing purposes.

File: Analytics/Defaults.cs (hunk 1 above)
Comment (lubird) [reply Analytics/Defaults.cs]: I assume we will be able to programmatically change it when we instantiate a new analytics object? For example, we might want to change the host to something else during initialization/configuration.
